Enable goa API endpoints as a fallback for the new API endpoints

rtnpro opened the pull request:
This change does the following things:
- Runs the new API endpoints at ``/api/``
- Run the goa API endpoints at ``/legacyapi/``
- Redirects from ``/api/*`` endpoint if route is missing to ``/legacyapi/``

This will enable a smooth migration from the Goa based API to the Gin based API, without any change in the client side.

rtnpro committed:
Re enable goa service.

rtnpro committed:
Run goa (/api/v1) and gin (/api/v2) endpoints simultaneously.

rtnpro committed:
Redirect api routes not found in new API to older goa API

rtnpro committed:
Added test case for api redirect to goa on missing route.

- Moved redirect logic from main.go to api/ pkg
- Refactor redirect logic to work with context.Request.URL.Path instead of context.Request.RequestURI as it's value is incosistent across running api server and tests

xcoulon commented on api/gin_engine_setup_test.go:
```diff
@@ -0,0 +1,70 @@
+)
+
+// Execute submits the request and returns the response recording fo subseauent verifications
+func Execute(s gormtestsupport.GinkgoTestSuite, r *http.Request) *httptest.ResponseRecorder {
```
we should avoid duplicating this utility method. What about moving this test in the `api/handler` pkg instead ? you could rename the `gin_engine_setup_test.go` file into `redirect_test.go` and verify that the call to `GET /api/foobar` is redirected, as you've already done in this PR. Also, could you verify that `POST`, `PATCH` and `DELETE` requests are also correctly handled ?

rtnpro replied:
Yes

rtnpro replied:
Fixed!

rtnpro committed:
Created a test lib for api tests.

rtnpro committed:
Added test cases for API redirection for POST, PATCH, DELETE

rtnpro commented:
@xcoulon I have updated the API redirection test cases for ``POST``, ``PATCH`` and ``DELETE`` request methods along with ``GET``.  Also, I have moved the common ``Execute`` function to ``api/test/utils.go`` so that it can be shared.

xcoulon commented:
great work, @rtnpro ! The code looks good overall, so I'll merge it. Thanks !
